### PR TITLE
table

### PR DESCRIPTION
**Pull request by BelohvostovAlex**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      tr.append(td3);
+      tr.append(td4);
+      tbody.append(tr);
+    }
+    document.body.append(table);
+
+    let allTd = document.querySelectorAll('td')
+    for(let i = 0; i < allTd.length; i++) {
+        allTd[i].style.cssText = `
+        padding: 10px`
```
Comment (igorTT) [inline script.js]: Broken spaces and missing semicolon

File: script.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        background: gray;
+        padding: 20px;
+        margin: 0 auto;
+        text-align: center;
+        border: 2px solid black;
+    `;
+    const caption = document.createElement("caption");
+    const thead = document.createElement("thead");
+    const trHead = document.createElement("tr");
+    const tbody = document.createElement("tbody");
```
Comment (igorTT) [inline script.js]: Make a helper function that gets tag name, styles, and text content as parameters and returns an element.

File: script.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    for (let user of users) {
+      let tr = document.createElement("tr");
+      let td1 = document.createElement("td");
+      let td2 = document.createElement("td");
+      let td3 = document.createElement("td");
+      let td4 = document.createElement("td");
+      td1.textContent = user.id;
+      td2.textContent = user.name;
+      td3.textContent = user.phone;
+      td4.textContent = user.email;
```
Comment (igorTT) [inline script.js]: replace with helpers

File: script.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+document.addEventListener("DOMContentLoaded", () => {
+  const getData = async () => {
+    const response = await fetch("https://jsonplaceholder.typicode.com/users");
+    const users = await response.json();
+    return users;
+  };
+
+  const createTable = async (thNames = ["id", "name", "phone", "website"]) => {
```
Comment (igorTT) [inline script.js]: You're trying to make this function reusable but, you have hard coded values on the cycle 48th line

File: script.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+document.addEventListener("DOMContentLoaded", () => {
+  const getData = async () => {
+    const response = await fetch("https://jsonplaceholder.typicode.com/users");
+    const users = await response.json();
+    return users;
```
Comment (igorTT) [inline script.js]: add try ... catch

File: script.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            font-size: 30px;
+            padding: 10px 0px
+        `;
+
+    table.append(caption);
+    table.append(thead);
+    table.append(tbody);
+    thead.append(trHead);
+
+    for (let i = 0; i < thNames.length; i++) {
```
Comment (igorTT) [inline script.js]: replace with foreach

**Review by igorTT — CHANGES_REQUESTED** [review]
Replace let with const where the values are immutable.

File: script.js [hunk 7]
```diff
@@ -1,74 +1,93 @@
...
+    try {
+      const response = await fetch(url);
+      const users = await response.json();
+      return users;
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const createEl = (el, text, styles) => {
```
Comment (igorTT) [inline script.js]: Pass arguments like an object. It'll help you to get rid of passing `null`, etc.

File: script.js [hunk 8]
```diff
@@ -1,74 +1,93 @@
...
+          font-size: 18px;
+          text-transform: uppercase;
+          padding: 10px 0px;
+          `
+        );
+        trHead.append(th);
+      });
 
-    let users = await getData();
+      const data = await fetchedData;
```
Comment (igorTT) [inline script.js]: Pass this argument as a callback function. It'll make the code less confusing.

File: script.js [hunk 9]
```diff
@@ -1,74 +1,93 @@
...
+          tr.append(td);
+        });
+        tbody.append(tr);
+      }
+      document.body.append(table);
 
-    let allTd = document.querySelectorAll('td')
-    for(let i = 0; i < allTd.length; i++) {
+      const allTd = document.querySelectorAll("td");
+      for (let i = 0; i < allTd.length; i++) {
```
Comment (igorTT) [inline script.js]: Replace with foreach

File: script.js [hunk 10]
```diff
@@ -1,74 +1,93 @@
...
+    tag.textContent = text;
+    tag.style.cssText = styles;
+    return tag;
   };
 
-  const createTable = async (thNames = ["id", "name", "phone", "website"]) => {
-    const table = document.createElement("table");
-    table.style.cssText = `
+  const createTable = async (tableCaption, thNames, fetchedData) => {
+    try {
```
Comment (igorTT) [inline script.js]: You don't need to catch errors in this function.

File: script.js [hunk 11]
```diff
@@ -1,74 +1,93 @@
...
-      const th = document.createElement("th");
-      th.style.cssText = `
-                font-size: 18px;
-                text-transform: uppercase;
-                padding: 10px 0px;
-            `;
-      th.textContent = thNames[i];
-      trHead.append(th);
-    }
+      thNames.forEach((element) => {
```
Comment (igorTT) [inline script.js]: Replace the element with a title

File: script.js [hunk 12]
```diff
@@ -1,74 +1,93 @@
...
+          const td = createEl("td", item[thName]);
+          tr.append(td);
+        });
+        tbody.append(tr);
+      }
+      document.body.append(table);
 
-    let allTd = document.querySelectorAll('td')
-    for(let i = 0; i < allTd.length; i++) {
+      const allTd = document.querySelectorAll("td");
```
Comment (igorTT) [inline script.js]: Why not add styles while creating components?

**Review by igorTT — CHANGES_REQUESTED** [review]

File: script.js (hunk 9 above)
Comment (BelohvostovAlex) [reply script.js]: deleted (useless)

File: script.js [hunk 13]
```diff
@@ -9,80 +9,71 @@ document.addEventListener("DOMContentLoaded", () => {
     }
   };
 
-  const createEl = (el, text, styles) => {
-    const tag = document.createElement(el);
-    tag.textContent = text;
-    tag.style.cssText = styles;
+  const createEl = (obj) => {
```
Comment (igorTT) [inline script.js]: Don't name entities with their type. Name it like args, config etc...
Use destructuring.

File: script.js [hunk 14]
```diff
@@ -9,80 +9,71 @@ document.addEventListener("DOMContentLoaded", () => {
...
 
   const createTable = async (tableCaption, thNames, fetchedData) => {
-    try {
-      const table = createEl(
-        "table",
-        null,
-        `
+      const table = createEl({
+        el: "table",
+        styles: `
```
Comment (igorTT) [inline script.js]: Fix the gaps.
Configure your code editor with formatting on save.

**Review by igorTT — APPROVED** [review]

**Pull request merged**